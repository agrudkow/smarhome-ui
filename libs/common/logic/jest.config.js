module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
