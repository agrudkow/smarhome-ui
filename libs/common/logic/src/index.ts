export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
