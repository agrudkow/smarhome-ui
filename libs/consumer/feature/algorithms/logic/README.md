# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
