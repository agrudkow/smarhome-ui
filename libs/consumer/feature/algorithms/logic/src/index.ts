export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
