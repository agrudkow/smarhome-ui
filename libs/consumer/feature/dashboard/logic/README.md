# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
