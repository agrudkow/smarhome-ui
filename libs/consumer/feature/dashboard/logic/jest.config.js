module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
