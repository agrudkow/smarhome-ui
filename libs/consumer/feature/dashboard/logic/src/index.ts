export * from './lib/use-billings';
