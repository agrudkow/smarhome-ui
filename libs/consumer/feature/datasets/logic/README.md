# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
