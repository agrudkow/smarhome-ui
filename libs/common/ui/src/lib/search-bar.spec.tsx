import React from 'react';
import { customRender } from '@smarthome/common/logic';
import SearchBar from './search-bar';
import { fireEvent } from '@testing-library/react';

describe(' SearchBar', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <SearchBar
        onSearch={jest.fn()}
        onInputValueChange={jest.fn()}
        inputValue=""
        inputPlaceHolder=""
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <SearchBar
        onSearch={jest.fn()}
        onInputValueChange={jest.fn()}
        inputValue=""
        inputPlaceHolder=""
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should set placeholder', () => {
    const { getByPlaceholderText } = customRender(
      <SearchBar
        onSearch={jest.fn()}
        onInputValueChange={jest.fn()}
        inputValue=""
        inputPlaceHolder="Test place holder"
      />
    );
    expect(getByPlaceholderText('Test place holder')).toBeTruthy();
  });

  it('should call handler function on input value change', () => {
    const handleChange = jest.fn();
    const { getByPlaceholderText } = customRender(
      <SearchBar
        onSearch={jest.fn()}
        onInputValueChange={handleChange}
        inputValue=""
        inputPlaceHolder="InputPlaceHolder"
      />
    );
    const input = getByPlaceholderText('InputPlaceHolder');
    fireEvent.change(input, { target: { value: 'test change' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('should call handler function on search click', () => {
    const handleSearch = jest.fn();
    const { getByText } = customRender(
      <SearchBar
        onSearch={handleSearch}
        onInputValueChange={jest.fn()}
        inputValue=""
        inputPlaceHolder="InputPlaceHolder"
      />
    );
    const input = getByText('Search');
    fireEvent.click(input);
    expect(handleSearch).toHaveBeenCalledTimes(1);
  });
});
