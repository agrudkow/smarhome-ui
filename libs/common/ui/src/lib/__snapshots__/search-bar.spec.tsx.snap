// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` SearchBar should match snapshot 1`] = `
.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 8px;
}

.c1 .MuiInput-underline:after {
  border-color: #1BBDCE;
}

.c3 {
  width: 100%;
  font-weight: 550;
  white-space: nowrap;
  background-color: #4E4E4E;
  color: #E3E2E1;
  opacity: 0.7;
  -webkit-transition: color 0.3s ease-in-out,background-color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out,background-color 0.3s ease-in-out;
}

.c3:hover {
  background-color: #1BBDCE;
  color: #102128;
}

.c0 {
  padding: 5px 15px;
  margin: 15px 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  width: auto;
  max-height: auto;
  max-width: auto;
  background-color: #fff;
  opacity: 1;
  border-radius: 6px;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  -webkit-flex-basis: 100px;
  -ms-flex-preferred-size: 100px;
  flex-basis: 100px;
  margin: auto;
}

@media (max-width:769px) {
  .c0 {
    margin: 5px 0;
  }
}

@media (max-width:480px) {
  .c0 {
    padding: 5px 5px;
  }
}

@media (max-width:480px) {
  .c2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="input-normal"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained c3 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Search
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
